discord_bots/models.py

```diff
@@ -205,6 +206,177 @@ class DiscordMember:
+    __tablename__ = "economy_donation"
+
+    id: str = field(
+        init=False,
+        default_factory=lambda: str(uuid4()),
+        metadata={"sa": Column(String, primary_key=True)},
+    )
+    sending_player_id: int = field(
+        metadata={
+            "sa": Column(BigInteger, ForeignKey("player.id"), nullable=True, index=True)
+        },
+    )
+    sending_player_name: str = field(
+        metadata={"sa": Column(String, nullable=True, index=True)},
+    )
+    receiving_player_id: int = field(
+        metadata={
+            "sa": Column(BigInteger, ForeignKey("player.id"), nullable=False, index=True)
+        },
+    )
+    receiving_player_name: str = field(
+        metadata={"sa": Column(String, nullable=False, index=True)},
+    )
+    value: int = field(metadata={"sa": Column(Integer, nullable=False)})
+
+    sending_player = relationship(
+        "Player",
+        foreign_keys=[sending_player_id.metadata['sa']],
+        back_populates="donation_senders",
+    )
+    receiving_player = relationship(
+        "Player", 
+        foreign_keys=[receiving_player_id.metadata['sa']],
+        back_populates="donation_receivers"
+    )
+    transactions = relationship("EconomyTransaction", back_populates="donation")
+
+
+@mapper_registry.mapped
+@dataclass
+class EconomyPrediction:
+    """
+    Stores prediction data on a game per player
+    """
+
+    __sa_dataclass_metadata_key__ = "sa"
+    __tablename__ = "economy_prediction"
+
+    id: str = field(
+        init=False,
+        default_factory=lambda: str(uuid4()),
+        metadata={"sa": Column(String, primary_key=True)},
+    )
+    player_id: int = field(
+        metadata={
+            "sa": Column(BigInteger, ForeignKey("player.id"), nullable=False, index=True)
+        },
+    )
+    player_name: str = field(
+        metadata={"sa": Column(String, nullable=False, index=True)},
+    )
+    finished_game_id: str = field(
+        metadata={
+            "sa": Column(
+                String, ForeignKey("finished_game.id"), nullable=True, index=True
+            )
+        },
+    )
+    in_progress_game_id: str = field(
+        metadata={
+            "sa": Column(
+                String, ForeignKey("in_progress_game.id"), nullable=True, index=True
+            )
+        },
+    )
+    team: int = field(metadata={"sa": Column(Integer, nullable=False, index=True)})
+    prediction_value: int = field(metadata={"sa": Column(BigInteger, nullable=False)})
+    outcome: bool = field(
+        metadata={"sa": Column(Boolean, nullable=True)},
+    )
+
+    player = relationship("Player", back_populates="prediction")
+    finished_game = relationship("FinishedGame", back_populates="prediction")
+    in_progress_game = relationship("InProgressGame", back_populates="prediction")
+    transactions = relationship("EconomyTransaction", back_populates="prediction")
+
+
+@mapper_registry.mapped
+@dataclass
+class EconomyTransaction:
```

Comment:
do you think we can store the resulting balance of the debitor/creditor here?

Reply:
added `new_balance` to EconomyTransaction model.
As currency totals are only tracked on player (not on games), I'm only updating this field of the source or destination account in `create_transaction` is of type Player.